### PR TITLE
feat: TC39 decorator proposal support

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit adds support for [TC39 Decorator Proposal](https://github.com/tc39/proposal-decorators).

These decorators are only available in transpiled sources - ie. non-JavaScript files
(because of lack of support in V8).

This entails that "experimental TypeScript decorators" are not available by default
and require to be configured, with a configuration like this:
```
{
  "compilerOptions": {
    "experimentalDecorators": true
  }
}
```

Closes https://github.com/denoland/deno/issues/19160

### CODE REVIEW

**Comment by bartlomieju** [comment]
Blocked by https://github.com/denoland/deno/pull/22033.

File: cli/args/mod.rs [hunk 1]
```diff
@@ -141,6 +141,7 @@ pub fn deno_registry_api_url() -> &'static Url {
 pub fn ts_config_to_emit_options(
   config: deno_config::TsConfig,
 ) -> deno_ast::EmitOptions {
+  // eprintln!("ts_config_to_emit_options options {:#?}", config);
```
Comment (littledivy) [inline cli/args/mod.rs:144]: Remove

File: cli/args/mod.rs [hunk 2]
```diff
@@ -157,11 +158,13 @@ pub fn ts_config_to_emit_options(
       "precompile" => (false, false, false, true),
       _ => (false, false, false, false),
     };
+  // eprintln!(
+  //   "ts_config_to_emit_options {}",
+  //   options.experimental_decorators
+  // );
```
Comment (littledivy) [inline cli/args/mod.rs:164]: Remove

File: cli/args/mod.rs [hunk 3]
```diff
@@ -1088,10 +1091,26 @@ impl CliOptions {
...
+    match result {
+      Ok(mut ts_config_for_emit) => {
+        if matches!(self.flags.subcommand, DenoSubcommand::Bundle(..)) {
+          // For backwards compatibility, force `experimentalDecorators` setting
+          // to true.
+          *ts_config_for_emit
+            .ts_config
+            .0
+            .get_mut("experimentalDecorators")
+            .unwrap() = serde_json::Value::Bool(true);
```
Comment (littledivy) [inline cli/args/mod.rs:1108]: Do we always have a `experimentalDecorators` field in `ts_config.0`? Might want to do:
```suggestion
            .entry("experimentalDecorators")
            .and_modify(|f| *f = serde_json::Value::Bool(true))
            .or_insert(serde_json::Value::Bool(true));
```

Comment (bartlomieju) [reply cli/args/mod.rs:1103]: Yes, this value is always filled by `deno_config`

**Review by littledivy — APPROVED** [review]
LGTM

**Comment by pokatomnik** [comment]
Thank you so much! I Appreciate this!

**Pull request merged**